zarr/_storage/v3.py

```diff
@@ -567,6 +567,16 @@ def _normalize_store_arg_v3(store: Any, storage_options=None, mode="r") -> BaseS
         return store
     if isinstance(store, os.PathLike):
         store = os.fspath(store)
+    if FSStore._fsspec_installed():
+        import fsspec
+        if isinstance(store, fsspec.FSMap):
+            return FSStoreV3(store.root,
+                             fs=store.fs,
+                             mode=mode,
+                             check=store.check,
+                             create=store.create,
+                             missing_exceptions=store.missing_exceptions,
+                             **(storage_options or {}))
```

Comment:
I think it's actually fine to pass `storage_options` through like this. If it is not empty, it will trigger an error via

https://github.com/zarr-developers/zarr-python/blob/1af77b63ad8d51a5a8dc2cd923bf73cb8abe5a64/zarr/storage.py#L1329-L1331

This is the desired behavior. Storage options will already be configured on the `FSMap`, so specifying them twice would not make sense.